adcm_aio_client/core/config/_objects.py

```diff
@@ -0,0 +1,469 @@
+    LocalConfigs,
+    MergeStrategy,
+)
+from adcm_aio_client.core.errors import ConfigComparisonError, RequesterError
+from adcm_aio_client.core.types import AwareOfOwnPath, WithRequesterProperty
+
+
+class ConfigOwner(WithRequesterProperty, AwareOfOwnPath, Protocol): ...
+
+
+# Config Entries Wrappers
+
+
+class _ConfigWrapper:
+    __slots__ = ("_name", "_schema", "_data")
+
+    def __init__(
+        self: Self,
+        name: LevelNames,
+        data: ConfigData,
+        schema: ConfigSchema,
+    ) -> None:
+        self._name = name
+        self._schema = schema
+        self._data = data
+
+
+class _Group(_ConfigWrapper):
+    __slots__ = ("_name", "_schema", "_data", "_wrappers_cache")
+
+    def __init__(self: Self, name: LevelNames, data: ConfigData, schema: ConfigSchema) -> None:
+        super().__init__(name, data, schema)
+        self._wrappers_cache = {}
+
+    def _find_and_wrap_config_entry[ValueW: _ConfigWrapper, GroupW: _ConfigWrapper, AGroupW: _ConfigWrapper](
+        self: Self,
+        item: AnyParameterName | tuple[AnyParameterName, type[ValueW | GroupW | AGroupW]],
+        value_class: type[ValueW],
+        group_class: type[GroupW],
+        a_group_class: type[AGroupW],
+    ) -> ValueW | GroupW | AGroupW:
+        if isinstance(item, str):
+            name = item
+        else:
+            name, *_ = item
+
+        level_name = self._schema.get_level_name(group=self._name, display_name=name)
+        if level_name is None:
+            level_name = name
+
+        cached_wrapper = self._wrappers_cache.get(level_name)
+        if cached_wrapper:
+            return cached_wrapper
+
+        parameter_full_name = (*self._name, level_name)
+
+        class_ = value_class
+        if self._schema.is_group(parameter_full_name):
+            class_ = a_group_class if self._schema.is_activatable_group(parameter_full_name) else group_class
+
+        wrapper = class_(name=parameter_full_name, data=self._data, schema=self._schema)
+
+        self._wrappers_cache[level_name] = wrapper
+
+        return wrapper
+
+
+class Value[T](_ConfigWrapper):
+    @property
+    def value(self: Self) -> T:
+        # todo probably want to return read-only proxies for list/dict
+        return self._data.get_value(parameter=self._name)
+
+    def set(self: Self, value: Any) -> Self:  # noqa: ANN401
+        self._data.set_value(parameter=self._name, value=value)
+        return self
+
+
+class _Desyncable(_ConfigWrapper):
+    def sync(self: Self) -> Self:
+        self._data.set_attribute(parameter=self._name, attribute="isSynced", value=True)
+        return self
+
+    def desync(self: Self) -> Self:
+        self._data.set_attribute(parameter=self._name, attribute="isSynced", value=False)
+        return self
+
+
+class DesyncableValue[T](_Desyncable, Value[T]):
+    def set(self: Self, value: Any) -> Self:  # noqa: ANN401
+        super().set(value)
+        self.desync()
+        return self
+
+
+class Group(_Group):
```

Comment:
as we have Group in RBAC it may lead to misunderstanding. may be use GroupParameter or GroupOfConfigParameter?

Reply:
As you wish, but for me it seems easier:
1. Let the user decide (when he wants Group from config and rbac both in the same code, he can rename one of them)
2. Name RBACGroup or UserGroup, etc.

IMO name that's used in typing shouldn't be too long

`GroupOfConfigParameter` => `DesyncableGroupOfConfigParameter` or `GroupOfConfigParameterOfHostGroup` (very long)